src/color/constants.ts

```diff
@@ -1,26 +1,107 @@
- * @field
- * @memberOf Color
+ * Also matching rgba(r g b / a) as per new specs
+ * https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb
+ * Formal syntax at the time of writing:
+ * <rgb()> =
+ *  rgb( [ <percentage> | none ]{3} [ / [ <alpha-value> | none ] ]? )  |
+ *  rgb( [ <number> | none ]{3} [ / [ <alpha-value> | none ] ]? )
+ * <alpha-value> = <number> | <percentage>
+ *
+ * For learners this is how you can read this regex
+ * Regular expression for matching an rgba or rgb CSS color value
+ *
+ * /^          # Beginning of the string
+ * rgba?       # "rgb" or "rgba"
+ * \(\s*       # Opening parenthesis and optional whitespace
+ * (\d{0,3}    # 0 to three digits R channel
+ *  (?:\.\d+)? # Optional decimal with one or more digits
+ * )           # End of capturing group for the first color component
+ * %?          # Optional percent sign after the first color component
+ * \s*         # Optional whitespace
+ * [\s|,]      # Separator between color components can be a space or comma
+ * \s*         # Optional whitespace
+ * (\d{0,3}    # 0 to three digits G channel
+ *  (?:\.\d+)? # Optional decimal with one or more digits
+ * )           # End of capturing group for the second color component
+ * %?          # Optional percent sign after the second color component
+ * \s*         # Optional whitespace
+ * [\s|,]      # Separator between color components can be a space or comma
+ * \s*         # Optional whitespace
+ * (\d{0,3}    # 0 to three digits B channel
+ *  (?:\.\d+)? # Optional decimal with one or more digits
+ * )           # End of capturing group for the third color component
+ * %?          # Optional percent sign after the third color component
+ * \s*         # Optional whitespace
+ * (?:         # Beginning of non-capturing group for alpha value
+ *  \s*        # Optional whitespace
+ *  [,/]       # Comma or slash separator for alpha value
+ *  \s*        # Optional whitespace
+ *  (\d{0,3}   # Zero to three digits
+ *    (?:\.\d+)? # Optional decimal with one or more digits
+ *  )          # End of capturing group for alpha value
+ *  %?         # Optional percent sign after alpha value
+ *  \s*        # Optional whitespace
+ * )?          # End of non-capturing group for alpha value (optional)
+ * \)          # Closing parenthesis
+ * $           # End of the string
+ *
+ * The alpha channel can be in the format 0.4 .7 or 1 or 73%
+ *
+ * WARNING this regex doesn't fail on off spec colors. it matches everything that could be a color.
+ * So the spec does not allow for rgba(30 , 45%  35, 49%) but this will work anyway for us
  */
-// eslint-disable-next-line max-len
-export const reRGBa =
-  /^rgba?\(\s*(\d{1,3}(?:\.\d+)?%?)\s*,\s*(\d{1,3}(?:\.\d+)?%?)\s*,\s*(\d{1,3}(?:\.\d+)?%?)\s*(?:\s*,\s*((?:\d*\.?\d+)?)\s*)?\)$/i;
+export const reRGBa = () =>
+  /^rgba?\(\s*(\d{0,3}(?:\.\d+)?%?)\s*[\s|,]\s*(\d{0,3}(?:\.\d+)?%?)\s*[\s|,]\s*(\d{0,3}(?:\.\d+)?%?)\s*(?:\s*[,/]\s*(\d{0,3}(?:\.\d+)?%?)\s*)?\)$/i;
```

Comment:
does it over complicate reusing some string e.g. `\(\s*(\d{0,3}(?:\.\d+)?%?)\s*[\s|,]\s*` in a string template?

Reply:
the issue with that is that i loose the ability to use the literal regex and i have to go with newRegexp.
I started with that if you look at the first commits. I got 400 extra bytes of code and i was pissed off.
I should try again now that i made the regex slightly different.